pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,156 @@ def vector_poly_data(orig, vec):
+                     [-3.6589393e-10, -3.0685656e-09, -1.0000000e+00]],
+                    dtype=float32)
+
+    Note that the principal axes have ones along the diagonal and zeros
+    in the off-diagonal. This indicates that the first principal axis is
+    aligned with the x-axis, the second with the y-axis, and third with
+    the z-axis. This is expected, since the mesh is already axis-aligned.
+
+    However, since the signs of the principal axes are arbitrary, the
+    first and third axes in this case have a negative direction.
+
+    Show the standard deviation along each axis.
+
+    >>> std
+    array([0.03060594, 0.01530297, 0.00714213], dtype=float32)
+
+    Convert the values to proportions for analysis.
+
+    >>> std / sum(std)
+    array([0.5769149 , 0.28845745, 0.1346276 ], dtype=float32)
+
+    From this result, we can determine that the axes explain approximately
+    58%, 29%, and 13% of the total variance in the points, respectively.
+
+    Let's compare this to the proportions of the known radii of the ellipsoid.
+
+    >>> radii / sum(radii)
+    array([0.6, 0.3, 0.1])
+
+    Note how the two ratios are similar, but do not match exactly. This is
+    because the points of the ellipsoid are prolate and are denser near the
+    poles. If the points were normally distributed, however, the proportions
+    would match exactly.
+
+    Create an array of normally distributed points scaled along the x-y-z axes.
+    Use the same scaling as the radii of the ellipsoid from the previous example.
+
+    >>> normal_points = rng.normal(size=(1000, 3))
+    >>> scaled_points = normal_points * radii
+    >>> axes, std = pv.principal_axes(scaled_points, return_std=True)
+    >>> axes
+    array([[-0.99997578,  0.00682346,  0.00136972],
+           [ 0.00681368,  0.99995213, -0.00702282],
+           [-0.00141757, -0.00701331, -0.9999744 ]])
+
+    Once again, the axes have ones along the diagonal as expected since the
+    points are already axis-aligned. Now let's examine the standard deviation
+    and compare the relative proportions.
+
+    >>> std
+    array([0.18798689, 0.0915765 , 0.03228786])
+
+    >>> std / sum(std)
+    array([0.60280948, 0.29365444, 0.10353608])
+
+    >>> radii / sum(radii)
+    array([0.6, 0.3, 0.1])
+
+    Since the points are normally distributed, the relative proportion of
+    the standard deviation matches the scaling of the axes almost perfectly.
+    """
+    points = _validation.validate_arrayNx3(points)
+
+    points_centered = points - np.mean(points, axis=0)
+    eig_vals, eig_vectors = np.linalg.eigh(points_centered.T @ points_centered)
+    axes = eig_vectors.T[::-1]  # columns, ascending order -> rows, descending order
+
+    # Ensure axes form a right-handed coordinate frame
+    if np.linalg.det(axes) < 0:
+        axes[2] *= -1
+
+    if return_std:
+        # Compute standard deviation and swap order from ascending -> descending
+        std = (np.sqrt(eig_vals) / len(points))[::-1]
```

Comment:
Division should be on the inside, because the covariance matrix is `points_centered.T @ points_centered / points.shape[0]`:
```suggestion
        std = (np.sqrt(eig_vals / len(points)))[::-1]
```
This should fix the doctest std results.

Perhaps we could put the division right up into the argument of `eigh()`, maybe even explicitly defining `covariance = ...`. It's not clear to me right now what is better for numerical stability. I suspect `eigh()` might be happier if its argument isn't potentially huge.

So perhaps worth putting the factor into the matrix. Perhaps we should define `points_centered = (points - points.mean(0)) / np.sqrt(points.shape[0])`, because then we still use the same memory footprint.

Reply:
> So perhaps worth putting the factor into the matrix. Perhaps we should define `points_centered = (points - 
points.mean(0)) / np.sqrt(points.shape[0])`, because then we still use the same memory footprint.

If I make this change (https://github.com/pyvista/pyvista/pull/6346/commits/9841419c4cf60308cb7157ab6b6f7e4cdd12cc40) the output ratios no longer match in `test_principal_axes_return_std`:
```
ratios_in
Out[1]: array([0.5       , 0.33333333, 0.16666667])
ratios_out
Out[2]: array([0.64285714, 0.28571429, 0.07142857])
```
So, this change is reverted. Maybe this falls in the realm of pre-mature optimization, so perhaps can be addressed in a separate PR only if/when its determined that numerical stability and/or memory usage are found to be an issue

> Division should be on the inside, because the covariance matrix is points_centered.T @ points_centered / points.shape[0]

Thanks, fixed in https://github.com/pyvista/pyvista/pull/6346/commits/4766baa1c07381329e0fae19f442cc12d5a7563e

Reply:
For what it's worth I tried moving the division by number of points into the argument of `eigh()` (so numerical errors would still be relevant in the calculation of the covariance matrix) or into the definition of `points_centered` (with which we'd "probably" minimize numerical issues). In all cases I get the exact same `std_out` for the `100_000``-sized normal blob in the `std` test (also the exact same axes).